src/controls/Control-handlers.spec.ts

```diff
@@ -0,0 +1,285 @@
+import { describe, it, expect, beforeEach, afterEach } from 'vitest';
+import { scalingXOrSkewingY, scalingYOrSkewingX } from './scaleSkew';
+import { Point } from '../Point';
+import type { FabricObject, TPointerEvent, Transform, TransformActionHandler } from '../../fabric';
+import { Canvas, controlsUtils, Rect } from '../../fabric';
+import { wrapWithFixedAnchor } from './wrapWithFixedAnchor';
+import { wrapWithFireEvent } from './wrapWithFireEvent';
+
+describe('fabric.controlsUtils', () => {
+  let eventData: TPointerEvent;
+  let transform: Transform & {
+    signX: number;
+    signY: number;
+    [key: string]: any;
+  };
+  const canvas = new Canvas(undefined);
+
+  beforeEach(() => {
+    const target = new Rect({ width: 100, height: 100 });
+    canvas.add(target);
+    eventData = {} as TPointerEvent;
+    transform = prepareTransform(target, 'mr');
+  });
+
+  afterEach(() => {
+    canvas.off();
+    canvas.clear();
+  });
+
+  function prepareTransform(
+    target: FabricObject,
+    corner: string,
+  ): Transform & { signX: number; signY: number } {
+    // TODO: this method is deprecated and jsdoc says that it should have disappeared before fabric 4?
```

Comment:
@asturur left a comment here in case you want to check it out

Reply:
Yeah that is a so dumb thing to solve.
When dragging a control, guessing where is the anchor point.
To be honest that should be explicit of the control that execute the action.
So if i put a control in a position x,y i should also say, the anchor point is w,z.

I plan to propose new controls for shapes ( same api, just different controls) so maybe that is the good occasion to also fix this.